modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -636,21 +636,10 @@ private void generateSupportingFiles(List<File> files, Map<String, Object> bundl
                 }
 
                 if (ignoreProcessor.allowsFile(new File(outputFilename))) {
-                    if (templateFile.endsWith("mustache")) {
-                        String template = readTemplate(templateFile);
-                        Mustache.Compiler compiler = Mustache.compiler();
-                        compiler = config.processCompiler(compiler);
-                        Template tmpl = compiler
-                                .withLoader(new Mustache.TemplateLoader() {
-                                    @Override
-                                    public Reader getTemplate(String name) {
-                                        return getTemplateReader(getFullTemplateFile(config, name + ".mustache"));
-                                    }
-                                })
-                                .defaultValue("")
-                                .compile(template);
-
-                        writeToFile(outputFilename, tmpl.execute(bundle));
+                    if (templateFile.endsWith(templatingEngine.getFileExtension())) {
```

Comment:
A key point that is missing here is:

```
compiler = config.processCompiler(compiler);
```

This allows generators to extend or configure the `Mustache.Compiler` instance. See example at https://github.com/OpenAPITools/openapi-generator/blob/ab9c4b5a611c37e380769390229ecb9d7ee9c92d/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractScalaCodegen.java#L146

Reply:
Yeah, 2nd option is definitely the easiest to implement, just pushed that. I think I'll be adding a `processTemplatingEngine` to the `CodegenConfig` interface, and *Codegen writers could do it themselves, casting to their particular language/generator preferred `TemplatingEngineAdpater`